src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -147,7 +147,7 @@ public static SdkResult GetSdkResult(SdkReference sdk, object nuGetVersion, SdkR
                         foreach (var result in results.Select(i => i.Result).Where(i => i.Success))
                         {
                             // Find the information about the package that was installed.  In some cases, the version can be different than what was specified (like you specify 1.0 but get 1.0.0)
-                            var installedPackage = result.GetAllInstalled().FirstOrDefault(i => i.Name.Equals(sdk.Name));
+                            var installedPackage = result.GetAllInstalled().FirstOrDefault(i => i.Name.Equals(sdk.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
Why not add an implicit operator that converts to LibraryIdentity? :) 

Then the equality of LibraryIdentity will account for the casing. 
The NuGet Framework equality will also account for 1.0 == 1.0.0.

Reply:
Okay I refactored things a bit to use `LibraryIdentity` in my second commit.  Please review it as I'm trusting you that it'll give me the comparison I need 😄